src/features/transactions/wrapEventSubscriber.ts

```diff
@@ -0,0 +1,53 @@
+import { Dispatch } from "@reduxjs/toolkit";
+
+import { Contract } from "ethers";
+
+import { store } from "../../app/store";
+import {
+  decrementBalanceBy,
+  incrementBalanceBy,
+} from "../balances/balancesSlice";
+import { SubmittedTransaction } from "./transactionsSlice";
+
+const handleWrapEvent = (eventName: string, data: any, dispatch: any) => {
+  const transactions = store.getState().transactions;
+
+  let transaction: SubmittedTransaction | null =
+    transactions.all.find((t: any) => t.hash === data[2].transactionHash) ||
+    null;
+
+  // If we don't have a 'transaction', we don't already know about this swap
+  // and therefore don't need to update the UI.
+  if (!transaction) return;
+
+  dispatch(
+    decrementBalanceBy({
+      //@ts-ignore
+      tokenAddress: transaction.order.senderToken,
+      //@ts-ignore
+      amount: transaction.order.senderAmount,
+    })
+  );
+  dispatch(
+    incrementBalanceBy({
+      //@ts-ignore
+      tokenAddress: transaction.order.signerToken,
+      //@ts-ignore
+      amount: transaction.order.senderAmount,
+    })
+  );
+};
+
+export default function subscribeToWrapEvents(params: {
+  wrapContract: Contract;
+  library: any;
+  dispatch: Dispatch;
+}) {
+  const { wrapContract, dispatch } = params;
+  wrapContract.on("Deposit", async (...data) =>
```

Comment:
`wrapContract.on("Deposit", async (data) =>`

Reply:
this is a problem, because if I don't spread it I don't get the data I want. All I get is an address
```Deposit 0xa3863513D0f2ECeDcd25ef3b5C73D67cb0aaD6B2```
vs 
```
Deposit 
Array(3) [ "0x5411c75107B2546b404A84ADBD555CBe3eAf4fE0", {…}, {…} ]
```

Reply:
Gotcha!